Observations with different type but same name will give a bad error message

If you have the following observations:
```
SUMMARY_OBSERVATION A {
VALUE = 0.1;
RESTART = 1;
KEY = FOPR;
};
GENERAL_OBSERVATION A {
DATA = RES;
RESTART = 1;
VALUE = 1;
};
```

then you get the following error message:
```
>               obs_vectors = dict(
                    **cls._handle_history_observation(
                        ensemble_config, conf_instance, std_cutoff, history, time_len
                    ),
                    **cls._handle_summary_observation(
                        ensemble_config, conf_instance, obs_time_list
                    ),
                    **cls._handle_general_observation(
                        ensemble_config, conf_instance, obs_time_list
                    ),
                )
E               TypeError: type object got multiple values for keyword argument 'A'
```

--- a/src/ert/parsing/new_observations_parser.py
+++ b/src/ert/parsing/new_observations_parser.py
@@ -1,4 +1,5 @@
 import os
+from collections import Counter
 from enum import Enum, auto
 from typing import Any, Dict, List, Literal, Tuple, TypedDict, Union
 
@@ -158,7 +159,7 @@ def _validate_conf_content(
             Tuple[ObservationType, FileContextToken, Dict[FileContextToken, Any]],
         ]
     ],
-):
+) -> ConfContent:
     result = []
     for decl in inp:
         if decl[0] == ObservationType.HISTORY:
@@ -186,7 +187,22 @@ def _validate_conf_content(
             )
         else:
             raise _unknown_declaration_error(decl)
+    _validate_unique_names(result)
     return result
+
+
+def _validate_unique_names(conf_content: ConfContent) -> None:
+    names_counter = Counter(n for _, n, *_ in conf_content)
+    duplicate_names = [n for n, c in names_counter.items() if c > 1]
+    errors = [
+        ErrorInfo(
+            f"Duplicate observation name {n}",
+            n.filename,
+        ).set_context(n)
+        for n in duplicate_names
+    ]
+    if errors:
+        raise ObservationConfigError.from_collected(errors)
 
 
 def _validate_history_values(

--- a/tests/test_config_parsing/observations_generator.py
+++ b/tests/test_config_parsing/observations_generator.py
@@ -124,13 +124,8 @@ class GeneralObservation(Observation):
         return "GENERAL_OBSERVATION"
 
 
-names = st.text(
-    min_size=1, max_size=8, alphabet=st.characters(min_codepoint=65, max_codepoint=90)
-)
-
-
 @st.composite
-def general_observations(draw, ensemble_keys, std_cutoff):
+def general_observations(draw, ensemble_keys, std_cutoff, names):
     kws = {}
     kws["data"] = draw(ensemble_keys)
     kws["name"] = draw(names)
@@ -157,7 +152,7 @@ positive_floats = st.floats(min_value=0.1, allow_nan=False, allow_infinity=False
 
 
 @st.composite
-def summary_observations(draw, summary_keys, std_cutoff):
+def summary_observations(draw, summary_keys, std_cutoff, names):
     kws = {}
     kws["name"] = draw(names)
     kws["key"] = draw(summary_keys)
@@ -186,11 +181,25 @@ def summary_observations(draw, summary_keys, std_cutoff):
 @st.composite
 def observations(draw, ensemble_keys, summary_keys, std_cutoff):
     assume(ensemble_keys is not None or summary_keys is not None)
+    names = st.text(
+        min_size=1,
+        max_size=8,
+        alphabet=st.characters(min_codepoint=65, max_codepoint=90),
+    )
+    seen = set()
+    unique_names = names.filter(lambda x: x not in seen).map(lambda x: seen.add(x) or x)
+    unique_summary_names = summary_keys.filter(lambda x: x not in seen).map(
+        lambda x: seen.add(x) or x
+    )
     observation_generators = []
     if ensemble_keys is not None:
-        observation_generators.append(general_observations(ensemble_keys, std_cutoff))
+        observation_generators.append(
+            general_observations(ensemble_keys, std_cutoff, unique_names)
+        )
     if summary_keys is not None:
-        observation_generators.append(summary_observations(summary_keys, std_cutoff))
+        observation_generators.append(
+            summary_observations(summary_keys, std_cutoff, unique_names)
+        )
         observation_generators.append(
             st.builds(
                 HistoryObservation,
@@ -206,7 +215,7 @@ def observations(draw, ensemble_keys, summary_keys, std_cutoff):
                     ),
                     max_size=2,
                 ),
-                name=summary_keys,
+                name=unique_summary_names,
             ),
         )
     return draw(

--- a/tests/test_config_parsing/test_enkf_obs_parsing.py
+++ b/tests/test_config_parsing/test_enkf_obs_parsing.py
@@ -913,3 +913,43 @@ def test_that_unknown_key_in_is_handled(tmpdir, observation_type):
         ert_config = ErtConfig.from_file("config.ert")
         with pytest.raises(ObservationConfigError, match="Unknown SMERROR"):
             _ = EnkfObs.from_ert_config(ert_config, new_parser=True)
+
+
+def test_validation_of_duplicate_names(
+    tmpdir,
+):
+    with tmpdir.as_cwd():
+        config = dedent(
+            """
+        NUM_REALIZATIONS 2
+
+        ECLBASE ECLIPSE_CASE
+        REFCASE ECLIPSE_CASE
+        OBS_CONFIG observations
+        """
+        )
+        with open("config.ert", "w", encoding="utf-8") as fh:
+            fh.writelines(config)
+        with open("observations", "w", encoding="utf-8") as fo:
+            fo.writelines(
+                """SUMMARY_OBSERVATION FOPR {
+                       KEY     = FOPR;
+                       RESTART = 1;
+                       VALUE   = 1.0;
+                       ERROR   = 0.1;
+                    };
+                    HISTORY_OBSERVATION FOPR;
+            """
+            )
+        with open("time_map.txt", "w", encoding="utf-8") as fo:
+            fo.writelines("2023-02-01")
+        run_sim(
+            datetime(2014, 9, 10),
+            [("FOPR", "SM3/DAY", None), ("FOPRH", "SM3/DAY", None)],
+        )
+
+        ert_config = ErtConfig.from_file("config.ert")
+        with pytest.raises(
+            ObservationConfigError, match="Duplicate observation name FOPR"
+        ):
+            _ = EnkfObs.from_ert_config(ert_config, new_parser=True)

